client/src/components/HabitForm/Content/Content.jsx

```diff
@@ -0,0 +1,154 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { ContentArea, Row } from '../HabitForm.styled';
+import FormGroup from '../../Shared/Forms/FormGroup';
+
+const Content = ({
+	step,
+	name,
+	tags,
+	description,
+	times,
+	period,
+	reminderEvery,
+	reminderTypes,
+	difficulty,
+	habitType,
+	habitStart,
+	onChange,
+}) => {
+	switch (step) {
+		case 0:
```

Comment:
Maybe create separate components for these cases instead of using the switch? For example, you can make Content stateful component and then change the currently displayed content through the state. It would be easier to manage the whole component that way in my opinion and would make the logic more encapsulated.
What I would do is to create Views directory inside `client/src/components/HabitForm` and I would create there all views through form can go as separate components (preferably stateless). 
You can then pass a function which can change current view to each component, alongside other needed data. This would also make each view much easier for writing unit tests.

 You can try to use the dropdown component that we already have in Shared folder but react-dropdown or similar library would be good too.

Also, I realize that this would require a significant refactor, so you're not in the mood for that it's not a big deal :D

Reply:
I've since done something similar, but with a switch just for for the views of the inner content of the form, which are split into different stateless components. This is definitely much better for unit testing.